src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,173 @@
+     * @return a Command object representing the command.
+     * @throws DukeException if the command cannot be parsed.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] commands = fullCommand.split("\\s");
+        String command = commands[0];
+        String args = commands.length > 1 ? String.join(" ", Arrays.copyOfRange(commands, 1, commands.length)) : "";
+
+        switch (command) {
+        case "bye":
+            if (!args.isBlank()) {
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+
+            return new ExitCommand();
+        case "list":
+            if (!args.isBlank()) {
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+
+            return new ListCommand();
+        case "due":
+            if (args.isBlank()) {
+                throw new DukeException("Date required for the due command.");
+            }
+
+            return new DueCommand(Parser.parseDate(args));
+        case "done":
+            if (args.isBlank()) {
+                throw new DukeException("Task number required for the done command.");
+            } else if (!args.chars().allMatch(Character::isDigit)) {
+                throw new DukeException("Only positive integers allowed for the done command.");
+            }
+
+            return new DoneCommand(Integer.parseInt(args));
+        case "delete":
+            if (args.isBlank()) {
+                throw new DukeException("Task number required for the delete command.");
+            } else if (!args.chars().allMatch(Character::isDigit)) {
+                throw new DukeException("Only positive integers allowed for the delete command.");
+            }
+
+            return new DeleteCommand(Integer.parseInt(args));
+        case "find":
+            if (args.isBlank()) {
+                throw new DukeException("Keyword cannot be blank.");
+            }
+
+            return new FindCommand(args);
+        case "todo":
+            return new AddCommand(TaskType.TODO, args);
+        case "deadline": {
+            String[] deadlineArgs = args.split("\\s/by\\s");
+            String description = deadlineArgs[0];
+            String by = deadlineArgs.length > 1
+                    ? String.join(" ", Arrays.copyOfRange(deadlineArgs, 1, deadlineArgs.length))
+                    : "";
+
+            return new AddCommand(TaskType.DEADLINE, description, Parser.parseDateTime(by));
+        }
+        case "event": {
+            String[] eventArgs = args.split("\\s/at\\s");
+            String description = eventArgs[0];
+            String at = eventArgs.length > 1
+                    ? String.join(" ", Arrays.copyOfRange(eventArgs, 1, eventArgs.length))
+                    : "";
+
+            return new AddCommand(TaskType.EVENT, description, Parser.parseDateTime(at));
+        }
+        default:
```

Comment:
Your switch structure looks good. There is no indentation for `case` clauses, and it comes with the default 👍